src/less/button/ds6/button.less

```diff
@@ -56,86 +95,29 @@ a.fake-btn {
         width: 1rem;
     }
 
-    &--small {
-        .button-size-rules(6px, 6px);
-
-        min-height: 32px;
-
-        &-fixed-height {
-            .button-size-rules(6px, 6px);
-
-            height: 32px;
-        }
-
-        &-truncated {
-            .button-size-rules(6px, 6px);
-            .btn-truncate;
-
-            height: 32px;
-        }
-    }
-
-    &--medium {
-        .button-size-rules(10px, 10px);
-
-        min-height: 40px;
-
-        &-fixed-height {
-            .button-size-rules(10px, 10px);
-
-            height: 40px;
-        }
-
-        &-truncated {
-            .button-size-rules(10px, 10px);
-            .btn-truncate;
-
-            height: 40px;
-        }
+    &--truncated {
```

Comment:
Just a notice to everyone @msendlakowski @seangates @agliga @DylanPiercey:

I'm not a fan of this LESS syntax of nesting modifier names using the ampersand symbol. It means we cannot search the source code for `btn--truncated` or `btn--large-truncated`. It's also a little harder to read, in the case of `btn--large-truncated I have to scan back upwards through the code to stitch together the relationships. I don't need to do this if the full selector name is right there in front of me. I've noticed this syntax creeping in to some of our code lately.

Reply:
Makes sense.  And I've had that "search" problem too.
It seems pretty pervasive in the code base.  I'll refactor the buttons to clean this up and add a backlog issue to address it in the other components later.